src/playlist-maker.js

```diff
@@ -1,7 +1,57 @@
 import videojs from 'video.js';
+import * as guid from './guid';
 import playItem from './play-item';
 import * as autoadvance from './auto-advance';
 
+/**
+ * Generate a unique id for each playlist item object. This id will be used to determine
+ * index of an item in the playlist array for cases where there are multiple items with
+ * the same source set.
+ *
+ * @private
+ *
+ * @param {Array}
+ *         An array of playlist items
+ *
+ * @return {Array}
+ *          The list of playlist items with unique ids
+ */
+const generatePlaylistItemId = (arr) => {
+  const list = [];
```

Comment:
This isn't returned, but it happens to work because the objects in `arr` are mutated. In other words, this `list` could simply be removed.